include/Type.hpp

```diff
@@ -75,6 +76,20 @@ namespace cytnx {
 
   }  // namespace internal
 
+  // helper metafunction to transform a variant into another variant via a
+  // transform template alias
+  template <typename V, template <typename> class Transform>
+  struct make_variant_from_transform;
+
+  template <template <typename> class Transform, typename... Args>
+  struct make_variant_from_transform<std::variant<Args...>, Transform> {
+    using type = std::variant<typename Transform<Args>::type...>;
```

Comment:
`using type = std::variant<typename Transform<Args>...>;` would enable us to transform the variant to the variant of the StorageImeplementation (typed storage).

To be compatible with `std`, we can use `make_variant_from_transform<std::variant<...>, std::add_pointer_t>`.

Reply:
Yes, that is why I put this metafunction in `Type.hpp`, it could be used in other places.